velox/exec/fuzzer/PrestoQueryRunner.cpp

```diff
@@ -554,46 +554,62 @@ std::optional<std::string> PrestoQueryRunner::toSql(
   return sql.str();
 }
 
-std::optional<std::string> PrestoQueryRunner::toSql(
-    const std::shared_ptr<const core::HashJoinNode>& joinNode) {
-  if (!isSupportedDwrfType(joinNode->sources()[0]->outputType())) {
-    return std::nullopt;
+static const std::string joinKeysToSql(
+    const std::vector<core::FieldAccessTypedExprPtr>& keys) {
+  std::stringstream out;
+  for (auto i = 0; i < keys.size(); ++i) {
+    if (i > 0) {
+      out << ", ";
+    }
+    out << keys[i]->name();
   }
+  return out.str();
+};
 
-  if (!isSupportedDwrfType(joinNode->sources()[1]->outputType())) {
-    return std::nullopt;
-  }
+static std::string filterToSql(const core::TypedExprPtr& filter) {
+  auto call = std::dynamic_pointer_cast<const core::CallTypedExpr>(filter);
+  return toCallSql(call);
+};
 
-  const auto joinKeysToSql = [](auto keys) {
-    std::stringstream out;
-    for (auto i = 0; i < keys.size(); ++i) {
-      if (i > 0) {
-        out << ", ";
-      }
-      out << keys[i]->name();
+static std::string joinConditionAsSql(const core::AbstractJoinNode& joinNode) {
+  std::stringstream out;
+  for (auto i = 0; i < joinNode.leftKeys().size(); ++i) {
+    if (i > 0) {
+      out << " AND ";
     }
-    return out.str();
-  };
+    out << joinNode.leftKeys()[i]->name() << " = "
+        << joinNode.rightKeys()[i]->name();
+  }
+  if (joinNode.filter()) {
+    if (!joinNode.leftKeys().empty()) {
+      out << " AND ";
+    }
+    out << filterToSql(joinNode.filter());
+  }
+  return out.str();
+};
 
-  const auto filterToSql = [](core::TypedExprPtr filter) {
-    auto call = std::dynamic_pointer_cast<const core::CallTypedExpr>(filter);
-    return toCallSql(call);
-  };
+std::string PrestoQueryRunner::getTableName(
+    const core::ValuesNodePtr& valuesNode) {
+  return fmt::format("t_{}", valuesNode->id());
+}
 
-  const auto& joinConditionAsSql = [&](auto joinNode) {
-    std::stringstream out;
-    for (auto i = 0; i < joinNode->leftKeys().size(); ++i) {
-      if (i > 0) {
-        out << " AND ";
-      }
-      out << joinNode->leftKeys()[i]->name() << " = "
-          << joinNode->rightKeys()[i]->name();
-    }
-    if (joinNode->filter()) {
-      out << " AND " << filterToSql(joinNode->filter());
-    }
-    return out.str();
-  };
+std::optional<std::string> PrestoQueryRunner::toSql(
```

Comment:
There is a bit of duplication here. Could we move this of part of this to the base class (assuming most of the SQL will be similar, if not all of it)?

Reply:
Done. I did the same for the toSql methods for values nodes and nested loop join nodes.